src/server/graph-generation/interaction/index.js

```diff
@@ -0,0 +1,139 @@
+const _ = require('lodash');
+const pc = require('../../pathway-commons');
+const logger = require('./../../logger');
+
+function edgeType(type) {
+  switch(type){
+    case 'in-complex-with':
+    case 'interacts-with':
+      return 'Binding';
+    case 'controls-phosphorylation-of':
+      return 'Phosphorylation';
+    case 'controls-expression-of':
+      return 'Expression';
+    default:
+      return '';
+  }
+}
+
+function getInteractionInfoFromPC(sources) {
+  const idArray = _.uniq(_.concat([], sources)); //convert sources to array
+
+  const geneIds = idArray.map(source =>
```

Comment:
We need to drop this altogether and just use `source` below. As it stands, this doesn't meet its goals/fragile. It seems to be trying to accept any URI as source map it to a gene ID...